framework/src/systems/NonlinearEigenSystem.C

```diff
@@ -188,16 +190,54 @@ NonlinearEigenSystem::postAddResidualObject(ResidualObject & object)
   }
 }
 
+void
+NonlinearEigenSystem::condenseOutConstraints()
+{
+  if (!dofMap().n_constrained_dofs())
+    return;
+
+  _eigen_sys.initialize_condensed_dofs();
+  const auto m = cast_int<numeric_index_type>(_eigen_sys.local_non_condensed_dofs_vector.size());
+  auto M = m;
+  _communicator.sum(M);
+  if (_eigen_sys.has_condensed_matrix_A())
+  {
+    _eigen_sys.get_condensed_matrix_A().init(M, M, m, m);
+    // A bit ludicrously MatCopy requires the matrix being copied to to be assembled
+    _eigen_sys.get_condensed_matrix_A().close();
+  }
+  if (_eigen_sys.has_condensed_matrix_B())
+  {
+    _eigen_sys.get_condensed_matrix_B().init(M, M, m, m);
+    _eigen_sys.get_condensed_matrix_B().close();
+  }
+  if (_eigen_sys.has_condensed_precond_matrix())
+  {
+    _eigen_sys.get_condensed_precond_matrix().init(M, M, m, m);
+    _eigen_sys.get_condensed_precond_matrix().close();
+  }
+}
+
 void
 NonlinearEigenSystem::solve()
 {
   const bool presolve_succeeded = preSolve();
   if (!presolve_succeeded)
     return;
 
+  std::unique_ptr<NumericVector<Number>> subvec;
```

Comment:
We are creating a temporary vector here and cannot use the existing solution vector in place?

Reply:
The existing vector may be used in-place. See https://petsc.org/main/manualpages/Vec/VecGetSubVector/